code/app.py

```diff
@@ -241,19 +246,21 @@ def stream_without_data(response):
 
 
 def conversation_without_data(request):
-    openai.api_type = "azure"
-    openai.api_base = f"https://{AZURE_OPENAI_RESOURCE}.openai.azure.com/"
-    openai.api_version = "2023-03-15-preview"
-    openai.api_key = AZURE_OPENAI_KEY
+    azure_endpoint = f"https://{AZURE_OPENAI_RESOURCE}.openai.azure.com/"
+    if AZURE_AUTH_TYPE == "rbac":
+        openai_client = AzureOpenAI(azure_endpoint=azure_endpoint,  api_version=AZURE_OPENAI_API_VERSION, azure_ad_token_provider=AZURE_TOKEN_PROVIDER)
```

Comment:
Extra space here. I think you need to run `black code` from a terminal. Or run `pre-commit install`. I ran this and there are 10 files that need formatting

Reply:
I've run black and then pushed!